Multiple op hang issue

umadevimcw opened the pull request:
### Ticket
#4702 

### Problem description
- Test was hanging and Row Major layout was used
- Few ops like, mul, add, sub, exp, gelu, tanh was hanging and failed as Row Major Layout was used. 

### What's changed
- Updated the layout of the test to TILE Layout 

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/11552992090)

umadevimcw committed:
#4702: Update test files

umadevimcw committed:
#4702: Change layout for the tests

umadevimcw committed:
#4702: Update layout for gelu test

Aswinmcw reviewed (approved):
LGTM

eyonland reviewed (changes requested):
Please update the documentation to reflect that we only support TILE here.
It seems to be missing on pointwise ops.  Example https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/gelu.html but I do see it on the binary ones.

umadevimcw commented:
> ###
> 1 review requesting changes and 2 approving reviews by reviewers with write access. [Learn more about pull request reviews.](https://docs.github.com/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews)
> ** 1 change requested **
> [<img alt="@eyonland" width="20" height="20" src="https://avatars.githubusercontent.com/u/41128502?s=40&amp;v=4">](/eyonland)
> ** eyonland ** requested c

Updated @eyonland

umadevimcw commented:
> Please update the documentation to reflect that we only support TILE here. It seems to be missing on pointwise ops. Example https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/gelu.html but I do see it on the binary ones.

Updated @eyonland

![Uploading Screenshot 2024-10-29 at 4.18.24 PM.png…]()

ayerofieiev-tt commented:
Reading the ticket, this change is.. hiding the issue???
Why does it hang? 
Why switching to Tiled layout helps? 
When will Row Major case be fixed?

ayerofieiev-tt commented:
Also if the op does not support the input, it must gracefully handle that.

dmakoviichuk-tt reviewed (changes requested):
Ill take a look on Monday. please don't merge.
It looks super strange.
In general we fix issue if tests fails but not the test :-D

umadevimcw commented:
@ayerofieiev-tt @dmakoviichuk-tt FYI, the #4702 was created few months back(before the new infra we have now) and team has done some analysis and added the comments as well https://github.com/tenstorrent/tt-metal/issues/4702#issuecomment-2084901737. 

With the new infra since we don't have RM support yet, now the test failed with Layout issue. Hence in this PR, updated the test files and documentation as well.
